### ISSUE TITLE
Game is improperly paused when minimzed during gameplay.

### ISSUE THREAD

**Issue by Tropix126** [ppy/osu#21599]
### Type

Game behaviour

### Bug description

If the game window loses focus while currently in gameplay, the resume overlay will not be shown when pressing "Continue". I'm unsure if this is intended or not (this is different behavior from osu!stable), but it does have some consequences (notably, the "Low Battery" modal on windows will trigger this behavior, causing this broken pause state). This affects all windowing modes, including fullscreen.

### Screenshots or videos

https://user-images.githubusercontent.com/42101043/206874918-461e936b-a7ef-4388-8a76-40933742ae56.mp4

### Version

2022.1208.0-lazer

### Logs

[performance-update.log](https://github.com/ppy/osu/files/10200917/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/10200918/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10200919/updater.log)
[database.log](https://github.com/ppy/osu/files/10200920/database.log)
[input.log](https://github.com/ppy/osu/files/10200921/input.log)
[network.log](https://github.com/ppy/osu/files/10200922/network.log)
[performance.log](https://github.com/ppy/osu/files/10200923/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/10200924/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/10200925/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/10200926/performance-input.log)

**Comment by peppy** [comment]
This is intended, as the cursor was not within the playfield. We only show the resume overlay currently when it is required to maintain persistence of cursor. Arguably, we could show it in all cases at the last location of the cursor that was within the playfield.

**Comment by Tropix126** [comment]
Makes sense. I do think that additional check would be beneficial though, since there are many where focus can be lost without user interaction.

**Comment by peppy** [comment]
Well the decision is whether those cases SHOULD have the resume cursor or not. It's a UX discussion.
